src/plugins/unified_search/public/query_string_input/fetch_index_patterns.ts

```diff
@@ -11,17 +11,30 @@ import { DataViewsContract } from '@kbn/data-views-plugin/public';
 
 export async function fetchIndexPatterns(
   indexPatternsService: DataViewsContract,
-  indexPatternStrings: string[]
+  indexPatternStrings: Array<{ type: 'title' | 'id'; value: string }>
 ) {
   if (!indexPatternStrings || isEmpty(indexPatternStrings)) {
     return [];
   }
 
-  const searchString = indexPatternStrings.map((string) => `"${string}"`).join(' | ');
+  const searchStringList: string[] = [];
+  const searchIdsList: string[] = [];
+  for (const { type, value } of indexPatternStrings) {
+    if (type === 'title') {
+      searchStringList.push(value);
+    } else {
+      searchIdsList.push(value);
+    }
+  }
+
+  const searchString = searchStringList.map((value) => `"${value}"`).join(' | ');
 
-  const exactMatches = (await indexPatternsService.find(searchString)).filter((ip) =>
-    indexPatternStrings.includes(ip.title)
-  );
+  const exactMatches = await Promise.all([
+    ...(
+      await indexPatternsService.find(searchString)
```

Comment:
should we "parallel" this? 
seems like now we first run this `find` and afterwards we resolve other by id